### DISCUSSION TITLE
Welcome to KiU-Net-pytorch Discussions!

### DISCUSSION THREAD

**Discussion by jeya-maria-jose** [jeya-maria-jose/KiU-Net-pytorch#18] [General]

**Comment by asadikine** [comment]
Hi, I'm working on vessel segmentation and I'd like to use KiU-Net for this task. I have some questions : 

1. By reading "Journal Extension" i notice that the number of features maps on the Table 8 are greater than (x2) features maps of KiU-Net your code (https://github.com/jeya-maria-jose/KiU-Net-pytorch/blob/master/arch/ae.py)! Can you clarify that point for me?
2. In the line 240 "self.final = nn.Conv2d(8,2,1,stride=1,padding=0)" Why did you use 2 for out_channels instead of 1 ? (error when using 2)
3. In skip connection, Why did you use add instead of concatenation ? (Perhaps to have a residual block !)

**Reply by jeya-maria-jose** [reply]
Hi,

1. The number of layers and number of filters are parameters that can be changed depending the size and resolution of the input images to trade-off with training time. So, even in our experiments, for some datasets, we would have used less number of filters (especially for 3D experiments). That is the reason why the number of feature maps might not be consistent in the code.
2. The output of the network is prediction maps for each class. Here, we have segmentation label (1) and background (0). So, the number is 2.
3. It is to reduce the compute. Concatenation can be viewed as a weighted addition where the weights are learned. Here , we assume equal weights and add them directly. https://stats.stackexchange.com/questions/361018/when-to-add-layers-and-when-to-concatenate-in-neural-networks
